tree/30_generic_methods/variable_string_from_command.cf

```diff
@@ -78,31 +78,40 @@ bundle agent variable_string_from_command(prefix, name, command)
       # define the variable within the prefix namespace
     pass2::
       "${prefix}.${name}" string =>  string_tail("${temp}", "${output_length}"),
-                                        ifvarclass => "returned_zero";
+                              if => "returned_zero";
 
   classes:
       "should_report"     expression => "${report_data.should_report}";
     pass1::
       "returned_zero"     expression => strcmp("${code}", "0");
     pass2::
-      "variable_defined" expression => isvariable("${prefix}.${name}");
+      "variable_defined"  expression => isvariable("${prefix}.${name}");
 
     any::
       "pass3" expression => "pass2";
       "pass2" expression => "pass1";
       "pass1" expression => "any";
 
+  files:
+    # avoid backups for tmp files
+    pass1.!pass2::
+      "${tmp_script_name}"
+        create        => "true",
+        edit_line     => insert_lines("${command}"),
+        edit_defaults => empty,
+        perms         => mog("700", "root", "0"),
+        comment       => "${class_prefix}",
+        action        => immediate_ignore_dry_run,
+        classes       => classes_generic("file_lines_present_${c_tmp_script_name}");
```

Comment:
you could even use  "inner_class_prefix", to remove the name of the method we don't use anymore